plugin/db/driver.go

```diff
@@ -492,3 +503,60 @@ func FormatParamNameInNumberedPosition(paramNames []string) string {
 	}
 	return fmt.Sprintf("WHERE %s ", strings.Join(parts, " AND "))
 }
+
+// IsAsterisksInTemplateValid checks if the pathTemplate is valid about asterisk.
+func IsAsterisksInTemplateValid(pathTemplate string) error {
+	const maxAsteriskNum = 2
+	if err := isMaxConsecutiveAsteriskValid(pathTemplate, maxAsteriskNum); err != nil {
+		return err
+	}
+	if err := isSingleAsteriskInTemplateValid(pathTemplate); err != nil {
+		return err
+	}
+	return isDoubleAsteriskInTemplateValid(pathTemplate)
+}
+
+// isSingleAsteriskInTemplateValid checks whether the single in file path template is valid.
+func isSingleAsteriskInTemplateValid(pathTemplate string) error {
+	return isMultipleTimesAsteriskInTemplateValid(pathTemplate, 1)
+}
+
+// isDoubleAsteriskInTemplateValid checks whether the consecutive double asterisks in file path template is valid.
+func isDoubleAsteriskInTemplateValid(pathTemplate string) error {
+	return isMultipleTimesAsteriskInTemplateValid(pathTemplate, 2)
+}
+
+// isMaxConsecutiveAsteriskValid returns true if the pathTemplate contains `n` consecutive asterisk at most.
+func isMaxConsecutiveAsteriskValid(pathTemplate string, n int) error {
+	re := regexp.MustCompile(strings.Repeat(`\*`, n+1))
+	if re.MatchString(pathTemplate) {
+		return errors.Errorf("path template %s contains more than %d asterisk", pathTemplate, n)
+	}
+	return nil
+}
+
+// isMultipleTimesAsteriskInTemplateValid checks whether the consecutive multiple asterisks in file path template is valid.
+// The rules are（）:
+// 1. consecutive multiple asterisks cannot be placed at the beginning or end.
+// 2. Both ends of consecutive multiple asterisks just can be * or /.
+// Take asteriskTimes = 2 as an example:
+// "**/test" and "test/**" will break the rule1.
+// "abc**" and "?**/" will break the rule2.
+func isMultipleTimesAsteriskInTemplateValid(pathTemplate string, asteriskTimes int) error {
```

Comment:
Can this function tell which rule the `pathTemplate` is breaking instead of an opaque "invalid xxx" error? The last thing a user wants to see for an error message is "this is wrong" without telling them what is actually wrong.

Reply:
SG. I can try to detect this.

Reply:
FYI @unknwon :
![CleanShot 2022-08-30 at 15 02 40](https://user-images.githubusercontent.com/87714218/187371660-387bcb1a-ce30-48b5-8e67-f0a296db3338.png)
![CleanShot 2022-08-30 at 15 04 08](https://user-images.githubusercontent.com/87714218/187371840-473c1a6a-1b1f-47c0-b758-e5f1d80655b1.png)
![CleanShot 2022-08-30 at 15 05 15](https://user-images.githubusercontent.com/87714218/187372019-beb80515-646e-4dc8-957c-3bbecb48f715.png)
![CleanShot 2022-08-30 at 15 05 46](https://user-images.githubusercontent.com/87714218/187372087-9d02efd1-a674-4eff-9b9a-ae2e09bef4ba.png)

Reply:
Nice 👍 thanks!